apps/docs.blocksense.network/components/common/Select.tsx

```diff
@@ -0,0 +1,268 @@
+    setIsOpen(false);
+  };
+
+  const toggleOpen = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (
+      selectRef.current &&
+      !selectRef.current.contains(event.target as Node)
+    ) {
+      setIsOpen(false);
+    }
+  };
+
+  const handleKeyDown = (event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      setIsOpen(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, []);
+
+  return (
+    <SelectContext.Provider
+      value={{
+        selectedValue,
+        onValueChangeAction: handleChange,
+        isOpen,
+        toggleOpen,
+      }}
+    >
+      <div ref={selectRef} className="select">
+        {children}
+      </div>
+    </SelectContext.Provider>
+  );
+};
+
+type SelectTriggerProps = HTMLAttributes<HTMLButtonElement> & {
+  className?: string;
+  children: ReactNode;
+  side: 'top' | 'bottom';
+};
+
+export const SelectTrigger = ({
+  className,
+  children,
+  side,
+  ...props
+}: SelectTriggerProps) => {
+  const context = useContext(SelectContext);
+  if (!context) {
+    throw new Error('SelectTrigger must be used within a Select');
+  }
+
+  const { toggleOpen, isOpen } = context;
+
+  return (
+    <Button
+      className={cn(
+        'select__trigger flex h-10 w-full items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground disabled:cursor-not-allowed disabled:opacity-50',
+        className,
+      )}
+      onClick={toggleOpen}
+      {...props}
+    >
+      {children}
+      {side === 'top' ? (
+        isOpen ? (
+          <Icon
+            className="p-0.5 dark:invert opacity-60"
+            size="xs"
+            icon={{
+              type: 'image',
+              src: '/icons/chevron-up.svg',
+            }}
+            ariaLabel="Up Arrow"
+          />
+        ) : (
+          <Icon
+            className="dark:invert opacity-55"
+            size="xs"
+            icon={{
+              type: 'image',
+              src: '/icons/chevron-down.svg',
+            }}
+            ariaLabel="Down Arrow"
+          />
+        )
+      ) : isOpen ? (
+        <Icon
+          className="dark:invert opacity-55"
+          size="xs"
+          icon={{
+            type: 'image',
+            src: '/icons/chevron-down.svg',
+          }}
+          ariaLabel="Down Arrow"
+        />
+      ) : (
+        <Icon
+          className="p-0.5 dark:invert opacity-60"
+          size="xs"
+          icon={{
+            type: 'image',
+            src: '/icons/chevron-up.svg',
+          }}
+          ariaLabel="Up Arrow"
+        />
+      )}
+    </Button>
```

Comment:
I don't see the need to change the chevrons based on the passed side, update them only according to the open/close state.

```suggestion
    {isOpen ? (
        <Icon
          className="p-0.5 dark:invert opacity-60"
          size="xs"
          icon={{
            type: 'image',
            src: '/icons/chevron-up.svg',
          }}
          ariaLabel="Up Arrow"
        />
      ) : (
        <Icon
          className="dark:invert opacity-55"
          size="xs"
          icon={{
            type: 'image',
            src: '/icons/chevron-down.svg',
          }}
          ariaLabel="Down Arrow"
        />
      )}
    </Button>
```